### PR TITLE
[Gekidou MM-45552] Replace Recent Search Item From MenuItem To OptionItem 

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
This PR converts the Recent Search item so it uses the `OptionItem` component

The `OptionItem` component has a new OptionType `REMOVE` which shows the close compass icon with attached `onClose` callback.

The design with MenuItem used a 40x40 area including the size 18 compassIcon. To implement this, I added the same size 18 icon with a surrounding hitslop of 11px on all sides.

The noticable difference will be when clicking a recent search item.  The UI response will be the same as the search modifiers.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45552

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Screenshots
image

#### Release Note

```release-note

```

### CODE REVIEW

File: app/components/option_item/index.tsx
Comment (enahum) [inline app/components/option_item/index.tsx:21]: I would call this onRemove to be consistent and self explanatory

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/components/option_item/index.tsx
Comment (jfrerich) [reply app/components/option_item/index.tsx:21]: Yeah, I was thinking about the name in general and if it should be even more generic than remove or close.  It's really an action icon onPress callback.  The function could do anything the caller wants.  Maybe make the name change and enhance if we decide to modify in the future?

**Review by matthewbirtch — APPROVED** [review]
Looks good to me @jfrerich.

I noticed that the 'Recent searches in {Team}' still shows if I've deleted all the recent searches. Can you create another ticket to remove this section if there are no recent searches?

**Comment by jfrerich** [comment]
> Looks good to me @jfrerich.
> 
> I noticed that the 'Recent searches in {Team}' still shows if I've deleted all the recent searches. Can you create another ticket to remove this section if there are no recent searches?

Ticket created -> https://mattermost.atlassian.net/browse/MM-45792

**Review by avinashlng1080 — APPROVED** [review]

**Review by larkox — APPROVED** [review]

**Comment by larkox** [comment]
@jfrerich heads up that there are conflicts in this PR

**Review by larkox — APPROVED** [review]

**Comment by larkox** [comment]
@avinashlng1080 How much your work on https://github.com/mattermost/mattermost-mobile/pull/6522 overlaps with this?

**Comment by avinashlng1080** [comment]
> @avinashlng1080 How much your work on #6522 overlaps with this?

Hmm, let's merge this one and then I'll rebase mine over it.

**Comment by avinashlng1080** [comment]
Hey @enahum , I see that this PR has got 3 approvals  but I can't merge it as it it still requires your review and the change you requested has been implemented.

Is this a branch security config?

**Review by enahum — APPROVED** [review]
Yes @avinashlng1080 branch is now secured, we can review the params, but I guess you could've dismiss my previous review

**Pull request merged**
